websocketOutportDriver/client/wsClient.go

```diff
@@ -19,70 +20,92 @@ var log = logger.GetOrCreate("wsClient")
 
 type client struct {
 	url                      string
-	blockingOnError          bool
+	blockingAckOnError       bool
 	retryDuration            time.Duration
 	wsConn                   WSConnClient
 	payloadParser            PayloadParser
-	operationHandler         OperationHandler
+	payloadProcessor         PayloadProcessor
 	uint64ByteSliceConverter sender.Uint64ByteSliceConverter
+	safeCloser               core.SafeCloser
 }
 
+// ArgsWsClient holds the arguments required to create a new websocket client handler
 type ArgsWsClient struct {
 	Url                      string
 	RetryDurationInSec       uint32
 	BlockingAckOnError       bool
-	OperationHandler         OperationHandler
+	PayloadProcessor         PayloadProcessor
 	PayloadParser            PayloadParser
 	Uint64ByteSliceConverter sender.Uint64ByteSliceConverter
 	WSConnClient             WSConnClient
 }
 
 // NewWsClientHandler will create a ws client to receive data from an observer/light client
-func NewWsClientHandler(args *ArgsWsClient) (*client, error) {
-	if args.OperationHandler == nil {
-		return nil, errNilOperationHandler
-	}
-	if args.PayloadParser == nil {
-		return nil, errNilPayloadParser
-	}
-	if args.WSConnClient == nil {
-		return nil, errNilWsConnReceiver
-	}
-	if args.Uint64ByteSliceConverter == nil {
-		return nil, errNilUint64ByteSliceConverter
-	}
-	if len(args.Url) == 0 {
-		return nil, errEmptyUrlProvided
+func NewWsClientHandler(args ArgsWsClient) (*client, error) {
+	err := checkArgs(args)
+	if err != nil {
+		return nil, err
 	}
 
 	urlReceiveData := url.URL{Scheme: "ws", Host: args.Url, Path: data.WSRoute}
 	retryDuration := time.Duration(args.RetryDurationInSec) * time.Second
 
 	return &client{
 		url:                      urlReceiveData.String(),
-		blockingOnError:          args.BlockingAckOnError,
+		blockingAckOnError:       args.BlockingAckOnError,
 		retryDuration:            retryDuration,
 		wsConn:                   args.WSConnClient,
 		payloadParser:            args.PayloadParser,
-		operationHandler:         args.OperationHandler,
+		payloadProcessor:         args.PayloadProcessor,
 		uint64ByteSliceConverter: args.Uint64ByteSliceConverter,
+		safeCloser:               closing.NewSafeChanCloser(),
 	}, nil
 }
 
+func checkArgs(args ArgsWsClient) error {
+	if args.PayloadProcessor == nil {
```

Comment:
for all the interfaces in this package we need to add IsInterfaceNil methods. Why? Testing directly for nil the argument which is an interface only solves half of the problem. For clarification, test this piece of code:
```GO
	t.Run("nil payload parser, should return error", func(t *testing.T) {
		args := createArgs()

		var nilPointer *testscommon.PayloadParserStub
		args.PayloadParser = nilPointer
		ws, err := NewWsClientHandler(args)
		require.Nil(t, ws)
		require.Equal(t, err, errNilPayloadParser)
	})
```
The nil check won't catch the nil argument but you will end up working with nil subcomponents, which, at some point, will panic the code.
Can be refactored in a new PR as this one got a little bit large.